stag-library-compiler/src/main/java/com/vimeo/stag/processor/StagProcessor.java

```diff
@@ -174,4 +172,29 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
 
         return true;
     }
+
+    /**
+     * Adds all classes annotated with {@link UseStag}
+     * to the supported type model. It does this recursively
+     * for unsupported types. Supported types handle their
+     * own enclosed element adding. Unsupported types that
+     * could be annotated are @interface and interface. Enums
+     * and classes are supported.
+     *
+     * @param useStagElement the element to add to the
+     *                       supported type model.
+     */
+    private static void processSupportedElements(@NotNull Element useStagElement) {
+        if (ElementUtils.isSupportedElementKind(useStagElement)) {
+            TypeMirror rootType = useStagElement.asType();
+            DebugLog.log("Annotated type: " + rootType + "\n");
+            SupportedTypesModel.getInstance().getSupportedType(rootType);
+            // TODO: Should we switch to adding all elements here or continue to let AnnotatedClass.initNestedClasses do its thing 2/2/17 [AR]
```

Comment:
Good point, I'll do that - me forgetting that even while writing that comment is probably good evidence that the logic should be combined